packages/@react-aria/utils/src/useResizeObserver.ts

```diff
@@ -26,18 +26,16 @@ export function useResizeObserver<T extends Element>(options: useResizeObserverO
       };
     } else {
       const resizeObserverInstance = new window.ResizeObserver((entries) => {
-        if (raf.current) {
+        if (raf.current || !Array.isArray(entries) || !entries.length) {
           return;
         }
         // avoid Error - ResizeObserver loop limit exceeded
         // it's ok to use a raf, ResizeObservers are already async and now we're just debouncing on frames
         raf.current = window.requestAnimationFrame(() => {
           raf.current = null;
-          if (!Array.isArray(entries) || !entries.length) {
-            return;
-          }
-          onResize();
         });
+        // call onResize immediately
+        onResize();
```

Comment:
wait so doesn't this re-introduce the issue fixed by #2891? Why would the `requestAnimationFrame` be needed at all now if it does nothing?

Reply:
sigh, you're right, it did re-introduce it. I'll revert the original PR and close this one until we can figure out something else.